### PR TITLE
feat: publish order and sell and repay

### PR DESCRIPTION
**Pull request by uzielflorida**

### CODE REVIEW

File: src/contracts/PurchaseBundler.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      type: 'tuple',
+      components: [
+        { name: 'loanId', type: 'uint256' },
+        { name: 'callbackData', type: 'bytes' },
+        { name: 'shouldDelegate', type: 'bool' },
+      ],
+    },
+  ];
+
+  static LoanRepaymentData = [
```
Comment (uzielflorida) [inline src/contracts/PurchaseBundler.ts:33]: TODO: move this to MSLV6

File: src/gondi.ts [hunk 2]
```diff
@@ -179,9 +193,16 @@ export class Gondi {
       tokenId,
       taker,
     };
-    const typedData = await this.api.generateNftOrderToBeSigned(orderInput);
-    const signature = await this.wallet.signTypedData(typedData);
-    return this.api.saveSignedNftOrder({ ...orderInput, signature });
+    let response = await this.api.publishOrder(orderInput);
+    while (response.__typename === 'SignatureRequest') {
```
Comment (mishuagopian) [inline src/gondi.ts:197]: I'd assume the order we expect here. I wouldn't iterate indefinitely.

File: src/gondi.ts [hunk 3]
```diff
@@ -179,9 +193,16 @@ export class Gondi {
...
+      const key = response.key as 'signature' | 'repaymentSignature';
+      orderInput[key] = await this.wallet.signTypedData(response.typedData as TypedDataDefinition);
+      response = await this.api.publishOrder(orderInput);
+    }
+
+    if (response.__typename !== 'SingleNFTOrder') throw new Error('This should never happen');
```
Comment (mishuagopian) [inline src/gondi.ts:203]: Not sure I'd leave this either

File: src/contracts/PurchaseBundler.ts (hunk 1 above)
Comment (mishuagopian) [reply src/contracts/PurchaseBundler.ts:32]: This should be read from abis. I'm pretty sure we can make it for this.

File: src/gondi.ts (hunk 2 above)
Comment (uzielflorida) [reply src/gondi.ts:197]: The number of signature depends if it's ask or not

**Review by mishuagopian — APPROVED** [review]

**Pull request merged**